src/domain/state.ts

```diff
@@ -205,20 +231,12 @@ export class ApplicationStateManager {
     throw new BadRequest('Operation not allowed');
   }
 
-  addDocument(
-    id: string,
-    name: string,
-    type: UploadDocumentType,
-    updatedBy: string,
-    isReviewer: boolean,
-  ) {
+  addDocument(id: string, name: string, type: UploadDocumentType, identity: Identity) {
     const current = this.currentApplication;
-    if (isReviewer && current.state !== 'APPROVED') {
-      throw new Error('not allowed');
-    }
+    checkAppIsApprovedAndUserCanAmend;
```

Comment:
looks like this should be a function call?

Reply:
oops sure should be, thx!